src/coreclr/src/tools/r2rdump/R2RDump.cs

```diff
@@ -78,13 +80,31 @@ public string FindAssembly(string simpleName, string parentFile)
                     string probeFile = Path.Combine(refPath, simpleName + extension);
                     if (File.Exists(probeFile))
                     {
-                        return probeFile;
+                        return Open(probeFile);
                     }
                 }
             }
 
             return null;
         }
+
+        private static unsafe MetadataReader Open(string filename)
+        {
+            byte[] Image = File.ReadAllBytes(filename);
+
+            fixed (byte* p = Image)
+            {
+                IntPtr ptr = (IntPtr)p;
+                PEReader peReader = new PEReader(p, Image.Length);
```

Comment:
`PEReader` constructor takes a pointer here and doesn't make a copy of the data, so what is returned by `Open()` will likely return garbage after the allocated `byte[] Image` has been collected by the GC.

Reply:
Tracked in https://github.com/dotnet/runtime/issues/1219